File: backend/app/schemas/commission.py
"""委托项目 Pydantic 模型."""

from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, ConfigDict


class MilestoneCreate(BaseModel):
    name: str
    due_date: Optional[str] = None
    description: Optional[str] = None
    order_index: int = 0


class MilestoneUpdate(BaseModel):
    status: Optional[str] = None
    due_date: Optional[str] = None
    description: Optional[str] = None
    name: Optional[str] = None
    order_index: Optional[int] = None


class MilestoneSchema(BaseModel):
    id: str
    commission_id: str
    name: str
    status: str
    due_date: Optional[str] = None
    description: Optional[str] = None
    order_index: int
    created_at: Optional[str] = None
    updated_at: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class PaymentCreate(BaseModel):
    amount: float
    method: Optional[str] = None
    milestone_id: Optional[str] = None
    notes: Optional[str] = None


class PaymentUpdate(BaseModel):
    status: Optional[str] = None
    paid_at: Optional[str] = None
    amount: Optional[float] = None
    method: Optional[str] = None
    notes: Optional[str] = None


class PaymentSchema(BaseModel):
    id: str
    commission_id: str
    milestone_id: Optional[str] = None
    amount: float
    method: Optional[str] = None
    status: str
    paid_at: Optional[str] = None
    notes: Optional[str] = None
    created_at: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class RevisionCreate(BaseModel):
    description: str
    client_feedback: Optional[str] = None
    files: Optional[list[str]] = None
    created_by: str = "artist"


class RevisionSchema(BaseModel):
    id: str
    commission_id: str
    description: str
    client_feedback: Optional[str] = None
    files: Optional[list[str]] = None
    created_by: str
    created_at: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class TimelineEvent(BaseModel):
    type: str  # milestone | payment | revision
    id: str
    title: str
    description: Optional[str] = None
    date: Optional[str] = None
    status: Optional[str] = None


class CalendarEvent(BaseModel):
    id: str
    title: str
    date: str
    type: str  # milestone_due | payment_received | revision_deadline


class DashboardStats(BaseModel):
    active_count: int = 0
    pending_payment: int = 0
    monthly_revenue: float = 0.0
    avg_ticket: float = 0.0

File: frontend/src/types/commission.ts
export interface CommissionProject {
  id: string
  user_id: string
  title: string
  description?: string
  client_name?: string
  status: string
  // Legacy / order-derived fields (may be populated from orders or payment_terms)
  amount?: number
  currency?: string
  due_date?: string
  priority?: string
  work_title?: string
  milestones?: Milestone[]
  payment_terms?: Record<string, unknown>[]
  orders?: CommissionOrder[]
  messages?: CommissionMessage[]
  order_count?: number
  message_count?: number
  created_at?: string
  updated_at?: string
}

export interface CommissionOrder {
  id: string
  project_id: string
  order_type: string
  amount: number
  status: string
  created_at?: string
}

export interface CommissionMessage {
  id: string
  project_id: string
  sender_id: string
  content: string
  created_at?: string
}

export interface Milestone {
  id: string
  commission_id: string
  name: string
  status: 'pending' | 'in_progress' | 'completed' | 'overdue'
  due_date?: string
  description?: string
  order_index: number
  created_at?: string
  updated_at?: string
}

export interface Payment {
  id: string
  commission_id: string
  milestone_id?: string
  amount: number
  method: string
  status: 'pending' | 'received' | 'partial' | 'overdue'
  paid_at?: string
  notes?: string
  created_at?: string
}

export interface Revision {
  id: string
  commission_id: string
  description: string
  client_feedback?: string
  files?: string[]
  created_by: string
  created_at?: string
}

export interface TimelineEvent {
  type: 'milestone' | 'payment' | 'revision'
  id: string
  title: string
  description?: string
  date?: string
  status?: string
}

export interface DashboardStats {
  active_count: number
  pending_payment: number
  monthly_revenue: number
  avg_ticket: number
}

export interface CalendarEvent {
  id: string
  title: string
  date: string
  type: 'milestone_due' | 'payment_received'
}

export interface CommissionFilters {
  user_id?: string
  status?: string
}
